diff --git res/ari/resource_channels.c res/ari/resource_channels.c
index e792a2a8d1..d1ee38a1c9 100644
--- res/ari/resource_channels.c
+++ res/ari/resource_channels.c
@@ -2101,7 +2101,7 @@ static void external_media_rtp_udp(struct ast_ari_channels_external_media_args *
 		0,
 		NULL,
 		args->app,
-		NULL,
+		args->data,
 		NULL,
 		0,
 		variables,
@@ -2151,7 +2151,7 @@ static void external_media_audiosocket_tcp(struct ast_ari_channels_external_medi
 		0,
 		NULL,
 		args->app,
-		NULL,
+		args->data,
 		NULL,
 		0,
 		variables,
